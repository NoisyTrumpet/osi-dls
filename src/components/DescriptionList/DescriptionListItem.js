import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from '../Icon';
import { Typography } from '../Typography';

const DescriptionListItem = ({ icon, titleText, descriptionText }) => (
  <>
    <dt className={`description-list-title ${icon ? 'has-icon' : 'no-icon'}`}>
      {icon && <Icon size={24} name={icon} addClass="description-list-icon" color="gray5" />}
      <Typography variant="body-medium">{titleText}</Typography>
    </dt>
    <dd className="description-list-description">
      <Typography variant="body" htmlTagOverride="p">{descriptionText}</Typography>
    </dd>
  </>
);

DescriptionListItem.propTypes = {
  icon: PropTypes.string,
  titleText: PropTypes.string.isRequired,
  descriptionText: PropTypes.string.isRequired,
};

export default DescriptionListItem;
