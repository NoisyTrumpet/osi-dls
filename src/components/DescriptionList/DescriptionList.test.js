import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DescriptionList } from './';
import DescriptionListItem from './DescriptionListItem';

describe('<DescriptionList />', () => {
  test('should render ', () => {
    const data = [
      {
        icon: 'icon',
        titleText: 'Front',
        descriptionText: 'You can do anything when you set your mind to it.',
      },
      {
        titleText: 'Inside',
        descriptionText: "Can't wait to see what this year brings!",
      },
      {
        icon: 'icon',
        titleText: 'Envelope & Seal',
        descriptionText: 'Includes one card and one envelope with a Gold Crown seal. Envelope color may vary.',
      },
    ];
    const wrapper = shallow(<DescriptionList descriptionListData={data} />);
    expect(wrapper.find(DescriptionListItem).length).toBe(3);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
