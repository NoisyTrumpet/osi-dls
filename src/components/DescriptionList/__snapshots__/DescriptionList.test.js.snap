// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DescriptionList /> should render  1`] = `
<dl
  className="description-list has-icons"
>
  <DescriptionListItem
    descriptionText="You can do anything when you set your mind to it."
    icon="icon"
    key="Front"
    titleText="Front"
  />
  <DescriptionListItem
    descriptionText="Can't wait to see what this year brings!"
    key="Inside"
    titleText="Inside"
  />
  <DescriptionListItem
    descriptionText="Includes one card and one envelope with a Gold Crown seal. Envelope color may vary."
    icon="icon"
    key="Envelope & Seal"
    titleText="Envelope & Seal"
  />
</dl>
`;
